### ISSUE TITLE
Iterate Find VA Forms Search Results

### ISSUE THREAD

**Issue by MickinSahni** [department-of-veterans-affairs/va.gov-team#11453]
## Issue Description
Update Find VA Forms - Search results to include new fields from Drupal (by way of the Forms API). 

---
## Tasks
- [x] Update frontend to consume new fields from Forms API
- [x] Update design to match specs of search results

Accessibility items (https://github.com/department-of-veterans-affairs/va.gov-team/issues/6825#issuecomment-669460219):
- [x] "Go to online tool" button will need an aria-label or SR-only text to be clear to screen readers. 
- [x] Download links should follow the accessible pattern VSA/VSP accessibility specialists collaborated on 
- [x] Individual buttons should include an accessible name that matches the displayed text.
- [x] Chevron: the > right angle bracket should also have a these attributes
- [x] The purpose of a link should be determined from the link text alone or from text together with programmatically determined link context 
- [x] Lighthouse API change has been made
- [x] Lighthouse API change has been tested and accepted by Lighthouse team

## Acceptance Criteria
- New form fields are available in search results

**Fields**
![image](https://user-images.githubusercontent.com/13204473/88711672-858eb880-d0e6-11ea-8b8e-2ec7ca843f28.png)

- "Revision date" field is now "Form last updated:"
- "Related to" is added.
- link to online tool
- link to form detail page `<-- these pages are now available (Jen 10/20)`

## Open Questions
- what the new response is gonna look like, but should be straightforward

## Create issues for:
- Any work Zach has left remaining

**Comment by MickinSahni** [comment]
Quick heads up - I included in the AC that the link to the form detail page should be included in the search results. 
That spec was included on the epic (#10758) but added to this implementation ticket cc @jenniferlee-dsva

**Comment by jenniferlee-dsva** [comment]
@MickinSahni - added a note for you ^ re just that. I'd be okay with consolidating this with form pages, but would prefer to do it in steps, since form pages have a lot of dependencies right now.

> link to form detail page `<-- this won't be available to work until form detail pages get published. @MickinSahni Should we pull this bullet item out for when form detail pages can be published? (Jen 7/28)`

**Comment by MickinSahni** [comment]
@jenniferlee-dsva yep works for me. i left a similar comment on our release plan ticket re: the timing of the links to the form detail pages from the search --> https://github.com/department-of-veterans-affairs/va.gov-team/issues/11311#issuecomment-665236879

which means analytics implementation, specifically calculating CTR, will be on hold until this ticket (iterating search results) is complete.

**Comment by ncksllvn** [comment]
**Todo** - Find Forms search results needs an additional field indicating whether a form has a published form detail pages. If it does, then the search result heading should be a hyperlink to the form detail page. If not, it should just be an ordinary heading. 

Tech steps -

1. Write the GraphQL query needed to determine whether a form has a detail page
1. Pass this GraphQL snippet to Lighthouse to implement and make sure it makes sense

**Comment by MickinSahni** [comment]
new ticket to be created in front of this ticket - @ncksllvn

**Comment by ncksllvn** [comment]
Ticket created here but I'll probably need to clarify it a bit, https://github.com/department-of-veterans-affairs/va.gov-team/issues/12027

**Comment by ncksllvn** [comment]
Some accessibility improvements/suggestions are mentioned here that would be good to implement with this work, https://github.com/department-of-veterans-affairs/va.gov-team/issues/6825#issuecomment-669460219

**Comment by ncksllvn** [comment]
Removing the "blocked" label, because the tech requirements have been met now - the Find Forms search API now includes a field on form detail pages, so once basic auth is lifted, we can now proceed with integrating the two (search results and detail pages.)

**Comment by MickinSahni** [comment]
@brianalloyd this just means we need to schedule the Sprint when we'll start working on the next iteration of www.va.gov/find-forms (after we launch form detail pages)

**Comment by ncksllvn** [comment]
Per this [thread](https://dsva.slack.com/archives/CUB5X5MGF/p1601307065012000), our `form_details_url` was removed from the API. After launching form detail pages, we will need to undo some changes in this PR to re-enable that property,  https://github.com/department-of-veterans-affairs/vets-api/pull/4989/files.

**Comment by MarciMcGuireGCIO** [comment]
Added discovery tag so that Zach and I can start researching what is required to complete these items.

**Comment by zacharymorel** [comment]
> Per this [thread](https://dsva.slack.com/archives/CUB5X5MGF/p1601307065012000), our `form_details_url` was removed from the API. After launching form detail pages, we will need to undo some changes in this PR to re-enable that property, [department-of-veterans-affairs/vets-api#4989](https://github.com/department-of-veterans-affairs/vets-api/pull/4989).

Just a reminder @MarciMcGuireGCIO , we will need some vets api work that will re enable per nicks' comment from above.

**Comment by zacharymorel** [comment]
Pinging @ncksllvn for this task listed: 
I think I need more context on this. What does this mean?
![Screen Shot 2020-11-27 at 10 38 57 AM](https://user-images.githubusercontent.com/26075258/100465158-d6b2b700-309c-11eb-8b2e-9405e6de3a44.png)

**Comment by ncksllvn** [comment]
@zacharymorel it's an a11y suggestion, referring to how a screen reader will read the link. I think this suggestion is related to the first point, `"Apply online" button will need an aria-label or SR-only text to be clear to screen readers.`.

Basically, the button that says `Apply online` might have markup like -

```html
<a href="download.pdf">Apply online <span class="vads-u-visibility--screen-reader">for form 10-10EZ</span></a>
```

The idea is that then the screenreader will read the whole string `Apply online for form 10-10ez` so that the link's purpose is clear without the visual presentation.

**Comment by zacharymorel** [comment]
Thank you, @ncksllvn .
I was looking at using aria-label attribute. Is there a preference to using `<a href="download.pdf">Apply online <span class="vads-u-visibility--screen-reader">for form 10-10EZ</span></a>` over something like
`<a href="download.pdf" aria-label="Apply online for form 10-10EZ">Apply online</span></a>`

**Comment by zacharymorel** [comment]
Nvm, I answered my own q. That attribute is to be used where the text is not viewable on the screen.

**Comment by ncksllvn** [comment]
That was an excellent question! I hadn't really thought of it before.

**Comment by jenniferlee-dsva** [comment]
@zacharymorel (FYSA @johnhashva @MarciMcGuireGCIO ) - please do not use "apply online" verbiage even in the screen reader alt text or aria labels here. 

The 'go to online tool' language is very deliberate because of the fact that not all of our forms and digital applications are "applying: for a benefit. Sometimes they will be managing task related actions such as adding or removing a dependent or uploading evidence for a claim, or transfer your g.il. bill to a family member, etc. (I.e., you don't "apply" to do these things online.)

**Comment by ncksllvn** [comment]
Oh yeah, I was just quoting off of the [original feedback](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6825#issue-580117989). I didn't peak back at the designs before commenting. Definitely do not use what I say as your source for requirements.

I _think_ this is the latest design file, https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/find-va-forms-redesign/design/VA-Form-search-landing-results.pdf. This ticket could have used some grooming I think, to help it make sense for someone new to the product.

**Comment by MarciMcGuireGCIO** [comment]
We're ready to meet re: Vets API.

**Comment by zacharymorel** [comment]
Self reference of a11y download link research docs: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152

**Comment by MarciMcGuireGCIO** [comment]
Adding a Collab Cycle issue as a blocker for completing this issue since there will be involvement from Lighthouse, QA, and UX.

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel you probably already know, but this one will need to be part of the Jan 12 go-live since we may need GA tag to track CTA clicks, a test case in TestRail, then we'll do internal staging review and request a review by QA and a11y.

**Comment by zacharymorel** [comment]
Does this need to be reviewed by QA?

**Comment by MarciMcGuireGCIO** [comment]
Yes, because we're adding a new feature.  I believe it has to go through the full collab cycle, but we can roll some other updates in with it.  I'm going to schedule a project kickoff to discuss so we know what is and is not required.

**Comment by zacharymorel** [comment]
So this is interesting to me. What counts to VSA as a new feature, what is the definition of a feature? 
To me this is just a few buttons/links added and some descriptive text about a search result.

**Comment by MarciMcGuireGCIO** [comment]
Based on my convo w/VSP, I believe it's anything that is a new iteration.  The other work didn't require full collab cycle because it was remediating things that were found in a QA and a11y review of the MVP that launched earlier this year.

**Comment by zacharymorel** [comment]
Okay, so we can self define a feature as anything that is being iterated over and/or brand new scope of work.

This should be merged and deployed soon.

**Comment by MarciMcGuireGCIO** [comment]
I believe so, yes.  I don't think we can merge and deploy this particular change until the GA tags are added there's a staging review, and I have to put in requests for those.  That was why I landed on Jan 12 due to the lead time each group requires + code freeze coming up.

**Comment by johnhashva** [comment]
Team -- this is a good use case for What's goes through CC/Review and what doesn't.   As far as I know, there is no specific guidance for this.  That said, we are confronting the same questions in Search -- and it can create swirl.   More clarity is needed and @MarciMcGuireGCIO let's team with @denisecoveyduc to get that for future reference.    My take for this ticket and its assortment of tasks (its a mish mash -- so that's problematic) none of this requires the Collab Cycle process -- these are iterative tasks that have AC and those responsible for validating that the AC has been met are the only necessary reviewers/checkpoints.

**Comment by MarciMcGuireGCIO** [comment]
@johnhashva Thank you.  

Agreed.  A more clear definition of the threshold for Collab Cycle would be helpful.

**Comment by jenniferlee-dsva** [comment]
Confirming ^ @johnhashva. Forms search and forms LP both separately went through design collab and intent reviews in fact.

**Comment by zacharymorel** [comment]
Some unforeseen issues have come up during development; I will need to rewrite our end to end (a.k.a system test) for find-forms in order to get this feature completed.

Luckily, I think we have enough padding in the estimation for it.

**Comment by MarciMcGuireGCIO** [comment]
Thanks for the heads-up.

**Comment by zacharymorel** [comment]
Refer to [this](https://dsva.slack.com/archives/CUB5X5MGF/p1607990657256300) conversation for some validation requests.

**Comment by MarciMcGuireGCIO** [comment]
Converted to an Epic since there are so many other dependencies and a Collab Cycle requirement

**Comment by zacharymorel** [comment]
@MarciMcGuireGCIO All is left is getting the two blockers of GA and API flag removed done. Once that is done, I can hook up the GA tags on the front end and I can then move this to validate!
